store/src/java/com/zimbra/cs/service/mail/SaveDraft.java

```diff
@@ -177,18 +182,20 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
                 if (delegateeAccountId != null) {
                     delegateeAccount = prov.getAccountById(delegateeAccountId);
                     oct = new OperationContext(delegateeAccount);
-                } else {
+                } else if (delegatorMbox != null) {
                     oct = new OperationContext(delegatorMbox);
                 }
-                mm = ParseMimeMessage.parseMimeMsgSoap(zsc, oct, delegatorMbox, msgElem, null, mimeData);
-                SendMsg.doSendMessage(oct, delegatorMbox, mm, mimeData.uploads, null, "r", null, null, false,
-                        false, false, false);
+                if (oct != null && delegatorMbox != null) {
+                    mm = ParseMimeMessage.parseMimeMsgSoap(zsc, oct, delegatorMbox, msgElem, null, mimeData);
+                    SendMsg.doSendMessage(oct, delegatorMbox, mm, mimeData.uploads, null, "r", null, null, false,
+                            false, false, false);
+                }
                 Element response = zsc.createElement(MailConstants.SAVE_DRAFT_RESPONSE);
```

Comment:
if above  SendMsg.doSendMessage doesn't executes whats the response here?

Reply:
This particular method i.e. SendMsg.doSendMessage is executed when we proxy the request to delegator Mbox. But the save draft call is actually triggered from Delegatee Mbox. So the response will be specific to Delegatee Mbox not delegator Mbox. That's why we are returning the empty response here and it doesn't depend on execution of SendMsg.doSendMessage.